dev/restinio/impl/connection.hpp

```diff
@@ -588,14 +621,19 @@ class connection_t final
 		void
 		consume_data( const char * data, std::size_t length )
 		{
-			auto & parser = m_input.m_parser;
+			auto * parser = &m_input.m_parser;
+
+			const auto parse_err =
+				llhttp_execute( parser, data, length );
+
+			const auto nparsed = [&]{
+				if( !parser->error_pos )
+					return length;
+				return static_cast< std::size_t >( parser->error_pos - data );
+			}();
 
-			const auto nparsed =
-				http_parser_execute(
-					&parser,
-					&( m_settings->m_parser_settings ),
-					data,
-					length );
+			assert( static_cast< std::size_t >( nparsed ) <= length );
```

Comment:
I think that if this case is possible then a check has to be present even in Release mode (where assertions are turned off). So I want to propose to use ordinary `if` statement with throwing an exception or to remove this assert at all.

Reply:
That must not happen. Parser declares that it will put either NULL or  position of the first char causing error.

But if llhttp-parser will put a garbage into `parser->error_pos` (it must not, but it's a software, so who knows) then we're in trouble.

That's what I know about the parser behavior. 
@eao197  So knowing the above:
 * add permanent check (with `trigger_error_and_close`);
 * leave it as debug check?

Reply:
I think that we either trust the tool (llhttp) and have no to do such checks or not trust and have to do checking even in release mode. Personally I don't like checks in debug builds because some errors occur only in release builds.

So let decide: do we trust to llhttp?

Reply:
Ok, I'll put a conventional check. 
It is not smth that would be called on every byte in normal case, so not an issue for performance.